util/annotations/annotations.go

```diff
@@ -146,6 +146,7 @@ var (
 
 	PossibleNonCounterInfo                  = fmt.Errorf("%w: metric might not be a counter, name does not end in _total/_sum/_count/_bucket:", PromQLInfo)
 	HistogramQuantileForcedMonotonicityInfo = fmt.Errorf("%w: input to histogram_quantile needed to be fixed for monotonicity (see https://prometheus.io/docs/prometheus/latest/querying/functions/#histogram_quantile) for metric name", PromQLInfo)
+	MixedFloatsHistogramsInfo               = fmt.Errorf("%w: encountered a mix of histograms and floats for", PromQLInfo)
```

Comment:
I think we don't even need a generic info-level error for the mixed case.
In case of aggregations with those collisions, we do warn-level already.
So this is only for BinOps, so let's be specific here already.

Reply:
Sorry, I could not get the point. Can you please elaborate it a little bit.

Reply:
We only need `MixedFloatsHistogramsInfo` for BinOps, so let's call it `MixedFloatsHistogramsBinOpInfo` already here and maybe add the kind of binops (as suggested for line 278). Maybe also avoid the "mix", as it is only for two arguments. How about:

```suggestion
	InvalidBinOpFloatHistogramInfo               = fmt.Errorf("%w: invalid operation between float and histogram for binary operator", PromQLInfo)
```